src/test/java/org/aszjch/demoapp/domain/articlefile/port/MinIOArticleFilePublisherIT.java

```diff
@@ -0,0 +1,63 @@
+package org.aszjch.demoapp.domain.articlefile.port;
+
+
+import io.minio.ListObjectsArgs;
+import io.minio.MinioClient;
+import io.minio.Result;
+import io.minio.messages.Item;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.aszjch.demoapp.TestcontainersConfiguration;
+import org.aszjch.demoapp.domain.articlefile.ArticleFile;
+import org.aszjch.demoapp.infrastructure.config.MinIOConnetionDetails;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Import(TestcontainersConfiguration.class)
+@SpringBootTest
+@Slf4j
+class MinIOArticleFilePublisherIT {
+
+    @Autowired
+    private ArticleFilePublisher minIOArticleFilePublisher;
+    @Autowired
+    private MinioClient minioClient;
+    @Autowired
+    private MinIOConnetionDetails minIOConnetionDetails;
+
+    @Test
+    void publishes_article_file_to_minio(@TempDir File tempDir) throws Exception {
+        String fileName = "test-file.txt";
```

Comment:
Złota zasada testów - jak najkrótsze. Jako zasadę "na oko" przyjmij sobie, że jak test ma więcej niż 7-10 linijek to coś jest nie tak. Tworzenie obiektów warto wynosić do prywatnych choćby metod. A jak tworzysz ich dużo na wskroś testów to tworzyć klasy dla nich.